schemas/task.yml

```diff
@@ -187,7 +193,6 @@ required:
   - provisionerId
   - workerType
   - schedulerId
-  - taskGroupId
```

Comment:
> In any case, I think I'll try to split it out so we have different required fields for inputs and outputs, but share the same data structure (due to reasons A and B listed in "Reason to change" section above).

Sadly this will add complexity for the code generators, since setting up the inheritance essentially requires using [allOf semantics](http://spacetelescope.github.io/understanding-json-schema/structuring.html). Although this is doable, it is a lot of extra work for code generators, so I think I'll leave it for now. :(